src/object/transform_cl.c

```diff
@@ -1511,17 +1511,15 @@ string_disk_size (const char *string)
  *    A jmp_buf has previously been established.
  */
 static char *
-get_string (OR_BUF * buf, int length)
+get_string (OR_BUF * buf, int length, DB_VALUE * value)
```

Comment:
One issue is that we mix db_value allocator  (currently db_private) with ws allocator (currently also db_private); however, this restricts the code flexibility (I assume there are many other codes with the same issue, but we can try to gradually fix it..).

I propose another solution which does not change signature:
Use local db_value; read db_value from from buffer without copy:
tp_VarNChar.data_readval (buf, value, &my_domain, length, false, NULL, 0);
return string = ws_copy_string ( db_get_string (value));
clear local db_value

similar for au_get_user_name (in that case we may have an memcpy overhead, but since it is client code we may accept it)

Reply:
Then callers should free the copied string. 
Which option you guys prefer to?

Reply:
Callers already free the copied strings for both functions (get_string and au_get_user_name)